flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java

```diff
@@ -194,26 +216,30 @@ public void run(SourceContext<T> sourceContext) throws Exception {
 		// all subtasks will run a fetcher, regardless of whether or not the subtask will initially have
 		// shards to subscribe to; fetchers will continuously poll for changes in the shard list, so all subtasks
 		// can potentially have new shards to subscribe to later on
-		fetcher = new KinesisDataFetcher<>(
-			streams, sourceContext, getRuntimeContext(), configProps, deserializer);
+		fetcher = createFetcher(streams, sourceContext, getRuntimeContext(), configProps, deserializer);
 
 		boolean isRestoringFromFailure = (sequenceNumsToRestore != null);
 		fetcher.setIsRestoringFromFailure(isRestoringFromFailure);
 
 		// if we are restoring from a checkpoint, we iterate over the restored
 		// state and accordingly seed the fetcher with subscribed shards states
 		if (isRestoringFromFailure) {
```

Comment:
@tony810430 could you briefly explain the changes here in this "if (isRestoringFromFailure)" block? I don't see how its relevant for the rescaling change. I may be missing something, though, so would be great if you explain a bit.